bundles/ui/org.eclipse.smarthome.ui/src/main/java/org/eclipse/smarthome/ui/internal/items/ItemUIRegistryImpl.java

```diff
@@ -476,6 +481,34 @@ public State getState(Widget w) {
     }
 
     /**
+     * Converts an item state to the type the widget supports (if possible)
+     *
+     * @param w - Widget in sitemap that shows the state
+     * @param s - State of the item
+     * @return the converted state or the original if conversion was not possible
+     */
+    private State convertState(Widget w, State s) {
+        State returnState = null;
+
+        if (w instanceof SwitchImpl) {
+            if (s instanceof PercentType) {
+                returnState = ((PercentType) s).as(OnOffType.class);
+            }
+        } else if (w instanceof SliderImpl) {
+            if (s instanceof PercentType) { // catches also HSBType etc
+                returnState = ((PercentType) s).as(PercentType.class);
+            }
+        }
```

Comment:
A `Colorpicker` only makes sense for a `Color` item and this does have a HSBType state anyway. Why would anyone want to attach a `ColorPicker` to a `Switch` or `Dimmer` item for example? What would be the use case? I don't see any, which is why I added no conversion for the `ColorPicker`.